### PR TITLE
Feature/stable loading

### PR DESCRIPTION
**Pull request by lbebber**
Adds:
- Export LoadingContext component, to allow data fetchers display a loading status while the data is being loaded.
- LoadingWrapper to LayoutContainer, which displays a loading status while the components are being loaded.

Changes:
- ExtensionPointComponent content is not rendered while the component is being loaded.
- Uses a loading bar at the top of the page in lieu of an entire-page loader while the layout info is being loaded

Works in tandem with a forthcoming PR on `vtex.store`

Test on: https://lbebber5--alssports.myvtex.com/

### CODE REVIEW

File: react/components/LoadingContext.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  const { isParentLoading } = useContext(LoadingContext)
+
+  const contentContainer = useRef<HTMLDivElement>(null)
+  const loaderContainer = useRef<HTMLDivElement>(null)
+
+  const loadingTimeout = useRef<NodeJS.Timer | null>(null)
+  const loadingComplete = useRef(false)
+
+  const updateLoading = useCallback(() => {
+    const areComponentsLoaded = Object.values(state.current.components).some(
```
Comment (jgfidelis) [inline react/components/LoadingContext.tsx]: is this name correct? Shouldn't it be named `areComponentsLoading`?

Comment (lbebber) [reply react/components/LoadingContext.tsx]: good catch!

File: react/components/LoadingContext.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  }, [isParentLoading, updateLoading])
+
+  const value = useMemo(
+    () => ({
+      setLoading,
+    }),
+    [setLoading]
+  )
+
+  const isSSR = !window.navigator
```
Comment (jgfidelis) [inline react/components/LoadingContext.tsx]: I would use canUseDOM here just for consistency

File: react/components/LoadingContext.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+export const withLoading = <T extends {}>(Component: ComponentType<T>) => {
+  const EnhancedComponent: FunctionComponent<T> = props => {
+    const isSSR = !window.navigator
+
+    if (isSSR) {
+      return <Component {...props} setLoading={() => {}} />
+    }
+
+    return (
+      <LoadingContext.Consumer>
```
Comment (jgfidelis) [inline react/components/LoadingContext.tsx]: use `useContext` here

File: react/components/RenderProvider.tsx [hunk 4]
```diff
@@ -962,7 +959,7 @@ class RenderProvider extends Component<Props, RenderProviderState> {
 
   private prefetchPages = () => {
     if (this.prefetchRoutes.size > 0) {
-      setTimeout(this.execPrefetchPages, 20 * 1000)
+      setTimeout(this.execPrefetchPages, 6 * 1000)
```
Comment (Klynger) [inline react/components/RenderProvider.tsx]: Why use `setTimeout` instead of idle event from window?

Comment (lbebber) [reply react/components/RenderProvider.tsx]: @Klynger good question, but this is just an adjustment that has hitched a ride on this PR. using the idle event might be good for another PR

**Pull request merged**